core/identity/src/dao/identity.rs

```diff
@@ -64,6 +64,14 @@ impl<'c> IdentityDao<'c> {
         .await
     }
 
+    pub async fn get_default_identity(&self) -> Result<Identity> {
+        use crate::db::schema::identity::dsl::*;
+        readonly_transaction(self.pool, |conn| {
+            Ok(identity.filter(is_default.eq(true)).first(conn)?)
+        })
+        .await
+    }
+
```

Comment:
You don't have to implement this by yourself, because there is `self.lock().await.get_default_id()` on IdentityService

Reply:
Yes and no, because `get_default_id()` is a method made on IdentityService and this wait function is not part of it (and I think that it shouldn't)